Packs/Devo/Integrations/Devo_v2/Devo_v2.py

```diff
@@ -155,25 +160,59 @@ def alert_to_incident(alert, user_prefix):
 
 
 # Monkey patching for backwards compatibility
+def _to_unix(date, milliseconds=False):
+    """
+    Convert date to a unix timestamp
+
+    date: A unix timestamp in second, a datetime object,
+    pandas.Timestamp object, or string to be parsed
+    by pandas.to_datetime
+    """
+
+    if date is None:
+        return None
 
+    elif date == 'now':
+        epoch = datetime.now().timestamp()
+    elif type(date) == str:
+        epoch = pd.to_datetime(date).timestamp()
+    elif isinstance(date, pd.Timestamp | datetime):
+        if date.tzinfo is None:
+            epoch = date.replace(tzinfo=timezone.utc).timestamp()
+        else:
+            epoch = date.timestamp()
+    elif isinstance(date, int | float):
+        epoch = date
+    else:
+        raise Exception('Invalid Date')
+
+    if milliseconds:
+        epoch *= 1000
 
-def get_types(self, linq_query, start, ts_format):
-    type_map = self._make_type_map(ts_format)
-    stop = self._to_unix(start)
-    start = stop - 1
+    return int(epoch)
 
-    response = self._query(
-        linq_query, start=start, stop=stop, mode="json/compact", limit=1
+
+def get_types(self, linq_query, start):
+    # Calculate stop time
+    stop = _to_unix(start)
+    start = stop - 10000
+
+    # Query the data
+    response = self.query(
+        query=linq_query,
+        dates={'from': timestamp_to_date(start), 'to': timestamp_to_date(stop)}
     )
 
     try:
         data = json.loads(response)
-        error_checking.check_status(data)
```

Comment:
How are we handling this error_checking related stuff ?

Reply:
I've added response validator.